### PR TITLE
[UUID] added unit tests 

### PR DESCRIPTION
**Pull request by Kirin-Chan**
Tested on macOS 10.12.2 / Xcode 8.2 and Ubuntu 16.10

### CODE REVIEW

File: TestFoundation/TestUUID.swift [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        XCTAssertEqual(children.count, 0)
+    }
+    
+    func test_bridgeToObjectiveC() {
+        let uuid    = UUID(uuid: test)
+        let bridged = uuid._bridgeToObjectiveC()
+        
+        XCTAssertNotNil(bridged)
+        XCTAssertTrue((bridged as Any) is NSUUID)
+        XCTAssertFalse((bridged as Any) is UUID)
```
Comment (phausler) [inline TestFoundation/TestUUID.swift]: this line fails when running this test on Darwin (not in swift-corelibs-foundation) so it probably means there is a bug somewhere.

File: TestFoundation/TestUUID.swift [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+        let uuid    = NSUUID(uuidString: testString)
+        let bridged = UUID._unconditionallyBridgeFromObjectiveC(uuid)
+        var bridgedConditional : UUID? = nil
+        let _ = UUID._conditionallyBridgeFromObjectiveC(uuid!, result: &bridgedConditional)
+        
+        XCTAssertNotNil(bridged)
+        XCTAssertNotNil(bridgedConditional)
+        XCTAssertTrue((bridged as Any) is UUID)
+        XCTAssertTrue((bridgedConditional as Any) is UUID)
+        XCTAssertFalse((bridged as Any) is NSUUID)
```
Comment (phausler) [inline TestFoundation/TestUUID.swift]: This also fails on the objective-c foundation so perhaps there is a failure somehow in the bridge code for these? or is the assert perhaps bad?

File: TestFoundation/TestUUID.swift [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+        XCTAssertFalse((bridgedConditional as Any) is NSUUID)
```
Comment (phausler) [inline TestFoundation/TestUUID.swift]: ditto here on the Darwin failure

Comment (Kirin-Chan) [inline TestFoundation/TestUUID.swift]: I will take a look at it.

**Comment by alblue** [comment]
This pull request has gone stale, closing.

**Pull request closed without merging**